micrometer-core/src/main/java/io/micrometer/core/instrument/binder/system/FileDescriptorMetrics.java

```diff
@@ -91,12 +111,26 @@ private double invoke(@Nullable Method method) {
 
     @Nullable
     private Method detectMethod(String name) {
+        if (osBeanClass == null) {
+            return null;
+        }
         try {
-            final Method method = osBean.getClass().getDeclaredMethod(name);
```

Comment:
`osBean.getClass()` will return the underlying implementation class, to which we don't have reflective access post-Jigsaw.